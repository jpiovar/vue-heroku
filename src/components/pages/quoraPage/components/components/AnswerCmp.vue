<template>
  <div class="card text-white bg-success p-1 mb-3" :id="itemData.id" :ref="itemData.id">
    <div class="card-header">
      <card-header
        :index="index"
        :itemData="itemData"
        :editedItemHeader="editedItem" :editedItemRef="editedItemHandle"
        :qId="qId"
        ref="cardHeader"
      />
    </div>
    <div class="card-body bg-light text-dark">
      <p class="card-text text-left">
        <card-body-text
          :itemData="itemData"
          :editedItemBody="editedItem"
          :saveItemRef="saveItemHandle"
        />
      </p>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { AnswerItem } from '../../../../../stores/quora/quora.types';
import CardHeader from '../../../../common/cardHeader/CardHeader.vue';
import CardBodyText from '../../../../common/cardBodyText/CardBodyText.vue';

@Component({
  components: {
    CardHeader,
    CardBodyText,
  },
  props: {
    index: {
      required: true,
      type: Number,
    },
    itemData: {
      required: true,
      type: Object,
    },
    qId: {
      required: true,
      type: String,
    },
  },
})
export default class AnswerCmp extends Vue {
  itemData!: AnswerItem;

  qId!: string;

  editedItem = { id: '', text: '' };

  editedItemHandle({ id, text }: { id: string, text: string }) {
    debugger;
    this.editedItem = { id, text };
  }

  saveItemHandle() {
    (this.$refs.cardHeader as any).saveItem();
  }
}
</script>

<style lang="scss" scoped>
</style>
