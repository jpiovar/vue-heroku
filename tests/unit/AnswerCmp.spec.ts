import { shallowMount, Wrapper } from '@vue/test-utils';
import AnswerCmp from '@/components/pages/quoraPage/components/components/AnswerCmp.vue';

const props = {
    index: 12,
    itemData: {id: 'a12'},
    qId: 'q12'
};

const getWrapperShallow = (propsData = props): Wrapper<AnswerCmp> => shallowMount(AnswerCmp, {
    propsData: {
        ...propsData
    },
});

describe('AnswerCmp.vue', () => {
    it('should match the snapshot', () => {
        const wrapper: Wrapper<AnswerCmp> = getWrapperShallow();
        expect(wrapper.vm.$el).toMatchSnapshot()
    });
    it('renders card-header', () => {
        const wrapper: Wrapper<AnswerCmp> = getWrapperShallow();
        const cardHeader = wrapper.find('.card-header');
        expect(cardHeader.exists()).toBe(true);
        expect(cardHeader.isVisible()).toBe(true);
    });
    it('renders card-body', () => {
        const wrapper: Wrapper<AnswerCmp> = getWrapperShallow();
        const cardBody = wrapper.find('.card-body');
        expect(cardBody.exists()).toBe(true);
        expect(cardBody.isVisible()).toBe(true);
    });
    it('renders component id with correct string', () => {
        const wrapper: Wrapper<AnswerCmp> = getWrapperShallow();
        expect(wrapper.attributes().id).toBe('a12');
    });
});