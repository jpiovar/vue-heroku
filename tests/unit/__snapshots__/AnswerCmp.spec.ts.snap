// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerCmp.vue should match the snapshot 1`] = `
<div
  class="card text-white bg-success p-1 mb-3"
>
  <div
    class="card-header"
  >
    <card-header-stub
      editeditemheader="[object Object]"
      editeditemref="function () { [native code] }"
      index="12"
      itemdata="[object Object]"
      qid="q12"
    />
  </div>
   
  <div
    class="card-body bg-light text-dark"
  >
    <p
      class="card-text text-left"
    >
      <card-body-text-stub
        editeditembody="[object Object]"
        itemdata="[object Object]"
        saveitemref="function () { [native code] }"
      />
    </p>
  </div>
</div>
`;
